### PR TITLE
feat: add functionality to filter Sidebar 

### LINKED ISSUE #10: adding functionality to filter Sidebar
here I have added functionality to filter sidebar to sort products according to choicce given by the user

### PR DESCRIPTION
**Pull request by Bhakti199**
Here I added functionality to Filter Sidebar #10  , so that user can sort the products according to given options. I used useReducer and useContext for this implementation.

Screenshot (1324)

### CODE REVIEW

**Review by munjalSaksham71 — APPROVED** [review]
Code is very well structured! Good work on it.

File: src/Components/FilterSidebar/FilterSidebar.jsx [hunk 1]
```diff
@@ -1,12 +1,29 @@
 import "./FilterSidebar.css";
 import { AiFillStar } from "react-icons/ai";
+import { useFilter } from "../../Context/index";
```
Comment (tushartiwari7) [inline src/Components/FilterSidebar/FilterSidebar.jsx:3]: you can use `"../../Context"` and it will work the same also looks good too.

**Review by Raghav888 — APPROVED** [review]
Good structure

File: src/Context/FilterContext/FilterProvider.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  const { productList } = useProductList();
+
+  const initialValue = {
+    sortByRange: 0,
+    sortByPrice: "",
+    sortByBrand: [],
+    includeOutOfStock: "include",
+    includeFastDelivery: "",
+    sortByRatings: "",
+  };
```
Comment (tushartiwari7) [inline src/Context/FilterContext/FilterProvider.js:17]: I liked the schema you used for Filters.

File: src/Context/FilterContext/FilterUtils.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
+const composeFunction = (state, functionList, data) =>
+  functionList.reduce((accumulator, currentFunction) => {
+    return currentFunction(accumulator, state);
+  }, data);
+
+const productListByRange = (productList, state) =>
+  state.sortByRange === 0
+    ? productList
+    : [...productList].filter((product) => product.price <= state.sortByRange);
```
Comment (tushartiwari7) [inline src/Context/FilterContext/FilterUtils.js:9]: suggestion: filter doesn't mutate the original array so there no need to apply filter on a copied array, you can directly apply to original array too

**Review by tushartiwari7 — APPROVED** [review]
You Used best practices and good naming's too, just some nitpickss have a look.

**Pull request merged**